src/Microsoft.ML.PCA/PcaTrainer.cs

```diff
@@ -461,6 +461,9 @@ internal PcaModelParameters(IHostEnvironment env, int rank, float[][] eigenVecto
             {
                 _eigenVectors[i] = new VBuffer<float>(eigenVectors[i].Length, eigenVectors[i]);
                 _meanProjected[i] = VectorUtils.DotProduct(in _eigenVectors[i], in mean);
+                Host.CheckParam(_eigenVectors[i].GetValues().All(FloatUtils.IsFinite),
```

Comment:
The same check is done when deserializing, so it's ok to also add it in the constructor:

https://github.com/dotnet/machinelearning/blob/a76936546f5e269fc1da61f33ab541389e445294/src/Microsoft.ML.PCA/PcaTrainer.cs#L506

The custom message was requested by the customer that opened the original issue on NimbusML.